homeassistant/components/version/sensor.py

```diff
@@ -1,156 +1,127 @@
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={ATTR_SOURCE: SOURCE_IMPORT}, data=config
+        )
     )
 
 
-class VersionData:
-    """Get the latest data and update the states."""
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up version sensors."""
+    coordinator: VersionDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    if (entity_name := entry.data[CONF_NAME]) == DEFAULT_NAME:
+        entity_name = entry.title
 
-    def __init__(self, api: HaVersion) -> None:
-        """Initialize the data object."""
-        self.api = api
+    version_sensor_entities: list[VersionSensorEntity] = [
+        VersionSensorEntity(
+            coordinator=coordinator,
+            entity_description=SensorEntityDescription(
+                key=str(entry.data[CONF_SOURCE]),
+                name=entity_name,
+            ),
+        )
+    ]
 
-    @Throttle(TIME_BETWEEN_UPDATES)
-    async def async_update(self):
-        """Get the latest version information."""
-        try:
-            await self.api.get_version()
-        except pyhaversionexceptions.HaVersionFetchException as exception:
-            _LOGGER.warning(exception)
-        except pyhaversionexceptions.HaVersionParseException as exception:
-            _LOGGER.warning(
-                "Could not parse data received for %s - %s", self.api.source, exception
-            )
+    async_add_entities(version_sensor_entities)
 
 
-class VersionSensor(SensorEntity):
-    """Representation of a Home Assistant version sensor."""
+class VersionSensorEntity(CoordinatorEntity, SensorEntity):
+    """Version sensor entity class."""
 
     _attr_icon = "mdi:package-up"
+    _attr_device_info = DeviceInfo(
+        name=f"{HOME_ASSISTANT} {DOMAIN.title()}",
+        identifiers={(HOME_ASSISTANT, DOMAIN)},
+        manufacturer=HOME_ASSISTANT,
+        entry_type=DeviceEntryType.SERVICE,
+    )
+
+    coordinator: VersionDataUpdateCoordinator
 
     def __init__(
         self,
-        data: VersionData,
-        description: SensorEntityDescription,
+        coordinator: VersionDataUpdateCoordinator,
+        entity_description: SensorEntityDescription,
     ) -> None:
-        """Initialize the Version sensor."""
-        self.data = data
-        self.entity_description = description
-
-    async def async_update(self):
-        """Get the latest version information."""
-        await self.data.async_update()
-        self._attr_native_value = self.data.api.version
-        self._attr_extra_state_attributes = self.data.api.version_data
+        """Initialize version sensor entities."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._attr_unique_id = (
+            f"{coordinator.config_entry.unique_id}_{entity_description.key}"
```

Comment:
Where do we set the config entry unique_id?

Reply:
We don't, this should use `entry_id`